chore: add docs for root execution context (#727)
mineiros-io/terramate: docs/codegen/generate-file.md

Each `generate_file` block requires a single label that is the path
where the generated file will be saved.
For more details about how code generation use labels check the [Labels Overview](overview.md#labels) docs.

The block has an optional **`context`** attribute which overrides the [generation context]
(overview.md#generation-context).

The **`content`** attribute defines the string that will be written on the file.

The value of the **`content`** has access to different Terramate features 
depending on the `context` defined.

For `context=root` it has access to:

> - Terramate Project Metadata references `terramate.root.*` and `terramate.stacks.*`

Reviewer: Maybe add a link to Terramate Functions, it will not have access to all tm_ functions, there we have the information of which is available etc.

Author: I did now but in the line below.

- Terramate function calls `tm_*(...)`
- Expressions using string interpolation `"${}"`

> and for `context=stack` (the default), it has access to everything defined above
> for `root` plus the features below:

Reviewer: ```suggestion
and for `context=stack` (the default), it has access to everything that `root` plus
have + the features below:
```

Reviewer: Just a suggestion, feel free to ignore.
